### PR TITLE
Getting CLASSPATH from Apache Ant build file.

### PR DESCRIPTION
**Pull request by taq**
To make this works, we just need a task called `classpath` on the Ant
build.xml file, that returns the classpath needed to compile the current
file on Vim. It can be a fixed string or a command that returns a
combination of paths, like:

```
<?xml version="1.0"?>
<project name="neomake" default="build" basedir=".">
   <path id="compile.path">
      <pathelement location="a.jar"/>
      <pathelement location="b.jar"/>
      <pathelement location="c.jar"/>
   </path>

   <target name="classpath">
      <echo message="${toString:compile.path}"/>
   </target>
</project>
```

which will print:

```
$ ant classpath -f build.xml -S -q
/tmp/a.jar:/tmp/b.jar:/tmp/c.jar
```
that can be used on Neomake.

### CODE REVIEW

File: autoload/neomake/makers/ft/java.vim [hunk 1]
```diff
@@ -348,5 +365,26 @@ function! s:GetGradleClasspath() abort
     return ''
 endf
 
+function! s:GetAntClasspath() abort
+    let ant = s:findFileInParent('build.xml', expand('%:p:h', 1))
+    if s:has_ant && filereadable(ant)
+        if !has_key(g:neomake_java_javac_ant_ftime, ant) || g:neomake_java_javac_ant_ftime[ant] != getftime(ant)
+            try
+                let f = tempname()
```
Comment (blueyed) [inline autoload/neomake/makers/ft/java.vim]: Unused var `f`.

File: autoload/neomake/makers/ft/java.vim [hunk 2]
```diff
@@ -348,5 +365,26 @@ function! s:GetGradleClasspath() abort
     return ''
 endf
 
+function! s:GetAntClasspath() abort
+    let ant = s:findFileInParent('build.xml', expand('%:p:h', 1))
```
Comment (blueyed) [inline autoload/neomake/makers/ft/java.vim]: Could use `neomake#utils#FindGlobFile` - but that's more an issue of the java module itself.

File: autoload/neomake/makers/ft/java.vim (hunk 1 above)
Comment (taq) [reply autoload/neomake/makers/ft/java.vim]: Argh, forgot to remove the `f` var after not using it. Neomake doesn't have a VimL lint? :-)

File: autoload/neomake/makers/ft/java.vim (hunk 2 above)
Comment (taq) [reply autoload/neomake/makers/ft/java.vim]: Yeah, I tried to follow the original methods.

**Comment by taq** [comment]
Removed the unused var.

File: autoload/neomake/makers/ft/java.vim (hunk 1 above)
Comment (blueyed) [reply autoload/neomake/makers/ft/java.vim]: Well, there is vint (which is fas, but does not detect thist), and vimlint (which is slow and therefore not enabled by default).
You also have to install some wrapper script for vimlint IIRC.

Comment (blueyed) [reply autoload/neomake/makers/ft/java.vim]: We run it on Travis though, so that's fine in general.

**Comment by taq** [comment]
Seems everything is fine now. :-)

**Comment by blueyed** [comment]
Waiting for feedback from @wsdjeg.

**Comment by wsdjeg** [comment]
I just test it, This PR works well for me. thanks for your work.

@blueyed I thinks this PR can be merged. :)

**Pull request merged**